src/main/java/org/folio/service/impl/AllowedServicePointsServiceImpl.java

```diff
@@ -49,107 +52,70 @@ private AllowedServicePointsResponse getForCreate(AllowedServicePointsRequest re
+      .recall(Set.copyOf(recall.values()));
   }
 
-  protected abstract Collection<String> getLendingTenants(AllowedServicePointsRequest request);
-
-  private boolean isAvailableInLendingTenant(AllowedServicePointsRequest request, String patronGroupId,
-    String tenantId) {
-
-    var allowedServicePointsResponse = getAllowedServicePointsFromLendingTenant(request,
-      patronGroupId, tenantId);
-    log.info("isAvailableInLendingTenant:: allowedServicePointsResponse: {}",
-      allowedServicePointsResponse);
+  private void combineAndFilterDuplicates(
+    Map<String, AllowedServicePointsInner> servicePoints, Set<AllowedServicePointsInner> toAdd) {
 
-    var availabilityCheckResult = Stream.of(allowedServicePointsResponse.getHold(),
-      allowedServicePointsResponse.getPage(), allowedServicePointsResponse.getRecall())
+    if (CollectionUtils.isEmpty(toAdd)) {
+      return;
+    }
+    toAdd.stream()
       .filter(Objects::nonNull)
-      .flatMap(Collection::stream)
-      .anyMatch(Objects::nonNull);
-
-    log.info("isAvailableInLendingTenant:: result: {}", availabilityCheckResult);
-    return availabilityCheckResult;
+      .forEach(t -> servicePoints.put(t.getId(), t));
   }
 
-  protected abstract AllowedServicePointsResponse getAllowedServicePointsFromLendingTenant(
+  protected abstract Collection<String> getLendingTenants(AllowedServicePointsRequest request);
+
+  protected abstract AllowedServicePointsResponse getAllowedServicePointsFromTenant(
     AllowedServicePointsRequest request, String patronGroupId, String tenantId);
 
   private AllowedServicePointsResponse getForReplace(AllowedServicePointsRequest request) {
     EcsTlrEntity ecsTlr = findEcsTlr(request);
-    final boolean requestIsLinkedToItem = ecsTlr.getItemId() != null;
-    log.info("getForReplace:: request is linked to an item: {}", requestIsLinkedToItem);
-
-    if (!requestIsLinkedToItem && isRequestingNotAllowedInLendingTenant(ecsTlr)) {
-      log.info("getForReplace:: no service points are allowed in lending tenant");
-      return new AllowedServicePointsResponse();
-    }
-
-    return getAllowedServicePointsFromBorrowingTenant(request);
-  }
-
-  private EcsTlrEntity findEcsTlr(AllowedServicePointsRequest request) {
-    final String primaryRequestId = request.getRequestId();
-    log.info("findEcsTlr:: looking for ECS TLR with primary request {}", primaryRequestId);
-    EcsTlrEntity ecsTlr = ecsTlrRepository.findByPrimaryRequestId(UUID.fromString(primaryRequestId))
-      .orElseThrow(() -> new EntityNotFoundException(String.format(
-        "ECS TLR for primary request %s was not found", primaryRequestId)));
-
-    log.info("findEcsTlr:: ECS TLR found: {}", ecsTlr.getId());
-    return ecsTlr;
-  }
-
-  private AllowedServicePointsResponse getAllowedServicePointsFromBorrowingTenant(
-    AllowedServicePointsRequest request) {
 
     log.info("getForReplace:: fetching allowed service points from borrowing tenant");
```

Comment:
We're not fetching them from the borrowing tenant anymore

Reply:
fixed